### ISSUE TITLE
Favorite action missing visual feedback

### ISSUE THREAD

**Issue by PVince81** [owncloud/web#1903]
### Steps to reproduce
1. Create a folder
2. Approach the cursor to the star
3. Click it
4. Wait

### Expected behaviour
When the cursor approaches it, it should transform to a hand. Or any other visual feedback appreciated to indicate that the star is clickable.
When clicking, a visual feedback should aknowledge the action, maybe spinner while the request is running, then eventually the star should turn yellow.

### Actual behaviour
No visual feedback on hover.
No visual feedback while the request is running. It looks like nothing happens until suddenly the star turns yellow after a few seconds.

### Versions
ownCloud Server 10.3.0 alpha2
Phoenix master 2f7b3d781175e63a45291127bb3d2dfc3cfa5153

**Comment by marcus-herrmann** [comment]
> No visual feedback while the request is running. It looks like nothing happens until suddenly the star turns yellow after a few seconds.

I would change the star's state *right after* the click, regardless of the request this click causes (independent from the *actual* favorite state). This would improve the perceived performance. Right now its display as starred is dependent from a state (`starred`) that's coming from a successful server call:

```
<oc-star class="uk-display-block" @click.native.stop="toggleFileFavorite(item)" :shining="item.starred" />
```
> No visual feedback on hover.

A `cursor: pointer` ("Mousehand") in CSS would not work with touch. Since it will become an actual button after the related PR is merged, the most robust solution would be to give the star button looks.

**Comment by pascalwengerter** [comment]
Fixed via #5000 right now on the `a11y-swarming` branch, coming to `master` soon
